lib/polyfill/patchedmediakeys_20140218.js

```diff
@@ -0,0 +1,800 @@
+  throw unsupportedKeySystemError;
+};
+
+
+/** @override */
+shaka.polyfill.PatchedMediaKeys.v20140218.MediaKeySystemAccess.prototype.
+    createMediaKeys = function() {
+  shaka.log.debug('v20140218.MediaKeySystemAccess.createMediaKeys');
+
+  // Alias
+  var v20140218 = shaka.polyfill.PatchedMediaKeys.v20140218;
+
+  var mediaKeys = new v20140218.MediaKeys(this.keySystem);
+  return Promise.resolve(/** @type {!MediaKeys} */ (mediaKeys));
+};
+
+
+/** @override */
+shaka.polyfill.PatchedMediaKeys.v20140218.MediaKeySystemAccess.prototype.
+    getConfiguration = function() {
+  shaka.log.debug('v20140218.MediaKeySystemAccess.getConfiguration');
+  return this.configuration_;
+};
+
+
+/**
+ * @this {!HTMLMediaElement}
+ * @override
+ * */
+shaka.polyfill.PatchedMediaKeys.v20140218.setMediaKeys = function(mediaKeys_) {
+  shaka.log.debug('v20140218.setMediaKeys');
+
+  // If mediaKeys is null, ensure mediaKeys on the videoElement is null
+  if (!mediaKeys_) {
+    delete this['mediaKeys'];
+    this['mediaKeys'] = null;
+
+    return;
+  }
+
+  // If mediaKeys equals videoElement's mediaKey instance, do nothing
+  if (this.mediaKeys == mediaKeys_) {
+    return;
+  }
+
+  // Alias
+  var v20140218 = shaka.polyfill.PatchedMediaKeys.v20140218;
+
+  // Remove any old listeners.
+  mediaKeys_.eventManager_.removeAll();
+
+  // Intercept and translate these prefixed EME events.
+  mediaKeys_.eventManager_.listen(this, 'msneedkey', v20140218.onMsNeedKey_);
+
+  delete this['mediaKeys'];  // in case there is an existing getter
+  this['mediaKeys'] = mediaKeys_;  // work around read-only declaration
+
+  // Wrap native HTMLMediaElement.msSetMediaKeys with Promise
+  try {
+    // IE11/Edge requires that readyState >=1 before mediaKeys can be set, so
+    // check this and wait for loadedmetadata if we are not in the correct state
+    if (this.readyState >= 1) {
+      return Promise.resolve(this.msSetMediaKeys(mediaKeys_.nativeMediaKeys_));
+    }
+    else {
+      return new Promise(function(resolve, reject) {
+        function setMediaKeysDeferred() {
+          this.msSetMediaKeys(mediaKeys_.nativeMediaKeys_);
+          this.removeEventListener('loadedmetadata', setMediaKeysDeferred);
+        }
+
+        this.addEventListener('loadedmetadata', setMediaKeysDeferred);
+
+        resolve();
+      }.bind(this));
+    }
+  } catch (exception) {
+    return Promise.reject(exception);
+  }
+};
+
+
+
+/**
+ * An implementation of MediaKeys.
+ *
+ * @constructor
+ * @param {string} keySystem
+ * @implements {MediaKeys}
+ */
+shaka.polyfill.PatchedMediaKeys.v20140218.MediaKeys = function(keySystem) {
+  shaka.log.debug('v20140218.MediaKeys');
+
+  this.keySystem_ = keySystem;
+  this.nativeMediaKeys_ = new MSMediaKeys(keySystem);
+
+  /** @private {!shaka.util.EventManager} */
+  this.eventManager_ = new shaka.util.EventManager();
+};
+
+
+/** @override */
+shaka.polyfill.PatchedMediaKeys.v20140218.MediaKeys.prototype.
+    createSession = function() {
```

Comment:
This takes an optional session type argument, which this polyfill needs to support in order to correctly report which session types it supports.  See the v01b polyfill for details.